### ISSUE TITLE
Regarding instart logic/g00 crap update

### ISSUE THREAD

**Issue by ghajini** [gorhill/uBO-Extra#119]
Test 1. tested with alone ublock origin on chrome
No g00 redirections/instart logic seen

`https://edmontonjournal.com/`   
`https://www.afterellen.com/`

`americanphotomag.com` now redirects to `https://www.popphoto.com/`

`https://www.baltimoresun.com/`
`https://www.boatingmag.com/`
`https://www.boston.com/`
`https://www.cafemom.com/`
`https://calgaryherald.com/`
`https://calgarysun.com/`
`capitalgazette.com`

`carrollcountytimes.com` redirects to `https://www.baltimoresun.com/maryland/carroll/`

`cattime.com`
`cbssports.com`
`chicagotribune.com`
`https://www.chron.com/`
`https://www.chroniclelive.co.uk/`

`citypaper.com` redirecs to `https://www.baltimoresun.com/citypaper/`

`https://www.comingsoon.net/` , admiral antiadblock

`computershopper.com` redirects to `pcmag.com`

`https://www.courant.com/`
`https://www.cruisingworld.com/`

`http://csgoutpost.com/ `site unavailable

`https://www.courant.com/`
`https://www.cycleworld.com/`
`https://www.dailydot.com/`
`https://www.dailypress.com`

`http://dayzdb.com/` site unreachable

`deathandtaxesmag.com`  redirects to `https://www.spin.com/death-and-taxes/`

`https://www.delmartimes.net/`

`destinationweddingmag.com`  redirects  to `https://www.islands.com/destination-weddings/`

`https://www.dirtrider.com/`
`https://dogtime.com/`

`http://dotaoutpost.com/` site unreachable

`https://download.cnet.com/windows/`
`https://edmontonjournal.com/`
`https://edmontonsun.com/`
`https://www.edmunds.com/`
`emedicinehealth.com`

`http://www.esohead.com/` site unreachable

`http://everquest.allakhazam.com/`
`extremetech.com`
`fieldandstream.com`
`financialpost.com`
`flyingmag.com`
`https://www.focus.de/`
`gamepedia.com`

`https://www.gamerevolution.com/`, admiral antiadblock

`gamespot.com`
`geek.com`
`gofugyourself.com`
`https://www.growthspotter.com/`

`hearthhead.com` redirects to`https://www.hearthstonetopdecks.com/`

`http://hockeysfuture.com/`  site unreachable

`hoylosangeles.com`
`ibtimes.com`
`ign.com`
`https://www.islands.com/`
`https://www.lajollalight.com/`
`latintimes.com`
`https://www.laptopmag.com/` 
`leaderpost.com`
`https://www.legacy.com/`
`https://www.lifewire.com/`
`livescience.com`

`lolking.ne`t redirects to `https://www.fanbyte.com/features/league-of-legends-model-viewer/#W1swLDAsMCwiMjY2IiwwLDBdXQ==`

`https://www.mamaslatinas.com/`
`marlinmag.com`
`mcall.com`
`medicaldaily.com`
`medicinenet.com`
`mmo-champion.com`
`momtastic.com`
`montrealgazette.com`
`https://www.motorcyclecruiser.com`
`motorcyclistonline.com`
`https://www.motortrend.com/`
`https://www.msn.com/en-in`

`https://musicfeeds.com.au` , admiral antiadblock

`mustangandfords.com`
`http://mysanantonio.com/`
`nasdaq.com`
`https://nationalpost.com/`
`newsarama.com`
`newsweek.com`
`orlandosentinel.com`
`ottawacitizen.com`
`https://ottawasun.com/`
`https://www.outdoorlife.com/`
`pcmag.com`

`https://www.playstationlifestyle.net/`  , admiral antiadblock

`popsci.com`
`https://www.ranchosantafereview.com/`
`https://www.range365.com/`
`https://www.ranker.com/`
`https://www.realclearpolitics.com/`

`realitytea.com` site unreachable

`redeyechicago.com` redirects to`https://www.chicagotribune.com/redeye/`

`salon.com`
`https://www.saltwatersportsman.com/`
`https://www.sandiegouniontribune.com/`
`https://www.seattlepi.com`
`https://www.saveur.com/`
`https://www.sfgate.com/`

`https://www.sherdog.com/`, admiral antiadblock

`slate.com`
`https://slickdeals.net/`

`southflorida.com` redirects to `https://www.sun-sentinel.com/entertainment/`

`https://www.space.com/` , antiadblock

`spin.com`
`sporcle.com`

`http://sportdiver.com/` site unreachable

`https://www.sportfishingmag.com/`
`http://sportingnews.com/`

`sportrider.com` redirects to`https://www.cycleworld.com/sport-rider/`

`spox.com`
`stereogum.com`
`https://www.sun-sentinel.com/`

`https://www.superherohype.com/` , admiral antiadblock

`tenplay.com.au` to `https://10play.com.au/`, antiadblock

`http://tf2outpost.com/` site unreachable

`thebalance.com`

`https://www.thefashionspot.com/` ,admiral antiadblock

`https://theprovince.com/`
`https://www.thespruce.com/`
`thestarphoenix.com`
`https://www.thoughtco.com/`
`timeanddate.com`
`https://www.timesunion.com/`
`tomsguide.com`
`https://www.tomsguide.fr/`
`https://www.tomshardware.co.uk/`
`tomshardware.com`
`tomshardware.de`
`https://www.tomshardware.fr/`
`https://torontosun.com/`

`https://www.totalbeauty.com/`,, admiral antiadblock

`trustedreviews.com`
`tvtropes.org`
`https://www.twincities.com/`
`https://vancouversun.com/`

`https://www.vg.no` gives antiadblock

`vibe.com`
`https://www.wakeboardingmag.com/`
`washingtonpost.com`

`waterskimag.com` redirects to `https://www.boatingmag.com/water-sports-boats/`

`https://www.webmd.com`
`wetteronline.de`

`wikia.com`redirects to `https://www.fandom.com/explore`

`https://www.windowscentral.com/`
`https://windsorstar.com/`
`winnipegsun.com`
`https://www.wowhead.com/
`
`wrestlezone.com` redirects to `https://www.mandatory.com/wrestlezone/` ,admiral antiablock triggered....

`https://www.xda-developers.com/`
`yachtingmagazine.com`

`zam.com` redirects to `https://corp.fanbyte.com/`

Test 2 . 
tested with ublock origin+ublock extra stable latest versions
 g00 redirection/instart logic seen

`atvrider.com`
`baggersmag.com`
`chowhound.com`
`https://www.diversitybestpractices.com/`
`https://www.floridatravellife.com/`
`goal.com`
`https://www.hotbikeweb.com/`
`infinitiev.com`
`metacritic.com`
`metrolyrics.com`
`streetchopperweb.com`
`superstreetbike.com`
`thoughtcatalog.com`
`tv.com`
`tvguide.com`
`utvdriver.com`
`wibc.com`
`workingmother.com`

**Comment by okiehsch** [comment]
The admiral sites are annoyances, update your filters and enable `uBO-annoyances`.

**Comment by Hrxn** [comment]
Guess what I just discovered... but luckily an issue is already open.

Can confirm for
```
metacritic.com
thoughtcatalog.com
```
both look completely broken for me right now.

uBO 1.21.6 + uBO-Extra 2.84 on Chrome (latest stable, Win64)

What happened, is it official that a new version of IL has been rolled out or something?

**Comment by okiehsch** [comment]
> What happened, is it official that a new version of IL has been rolled out or something?

https://github.com/NanoMeow/QuickReports/issues/1640#issuecomment-519753460

**Comment by ghajini** [comment]
It remains unanswered, why ubo extra is running on pages that don't have g00/instart logic

**Comment by rpodric** [comment]
So, when certain sites are removed from uBO-Extra, such as some of the ones mentioned above were (e.g. tvguide.com, which was missing things of late), and as a result now more or less feel like they're running without any blocking at all (i.e. lots of ads coming through), what should be the course of action?  Working to see what can be done over at EasyList?  Such sites were in uBO-Extra in the first place for a good reason, so I'm thinking there may not be much to be done, at least in some cases.

**Comment by okiehsch** [comment]
Interestingly the removed domains I checked  are back to using the May version of Instart Logic on my end.
Can you check `tvguide.com` on your end?

Go to `view-source:https://www.tvguide.com`
My end:
![image](https://user-images.githubusercontent.com/16838044/64315625-dbc8a980-cfb2-11e9-9fda-98578a2c24b9.png)

Is it the same on your end?
If so I could re-add the domains that use that version.
`uBO-extra` works fine on such sites.

**Comment by rpodric** [comment]
There's no mention of that in the source of that site at all here, in Chrome or Firefox. Is there some trick to it?

**Comment by okiehsch** [comment]
Um, but you do get redirected if you go to `https://www.tvguide.com`?

**Comment by okiehsch** [comment]
You can check if you get redirected by looking at your logger output. If you get redirected by Instart Logic you should see a lot of requests with `g00` in it.
![Unbenannt](https://user-images.githubusercontent.com/16838044/64318311-11bd5c00-cfba-11e9-9d43-e4e160e90b6b.JPG)

**Comment by gwarser** [comment]
No IL on my side.

<details>
<summary>Details</summary>

```
<!DOCTYPE html>
--
  | <html lang="en">
  | <head>
  | <meta charset="utf-8" />
  |  
  | <script type="text/javascript">(window.NREUM\|\|(NREUM={})).loader_config={xpid:"XQcAUFNADAYCUVhTDw=="};window.NREUM\|\|(NREUM={}),__nr_require=function(t,n,e){function r(e){if(!n[e]){var o=n[e]={exports:{}};t[e][0].call(o.exports,function(n){var o=t[e][1][n];return r(o\|\|n)},o,o.exports)}return n[e].exports}if("function"==typeof __nr_require)return __nr_require;for(var o=0;o<e.length;o++)r(e[o]);return r}({1:[function(t,n,e){function r(t){try{s.console&&console.log(t)}catch(n){}}var o,i=t("ee"),a=t(18),s={};try{o=localStorage.getItem("__nr_flags").split(","),console&&"function"==typeof console.log&&(s.console=!0,o.indexOf("dev")!==-1&&(s.dev=!0),o.indexOf("nr_dev")!==-1&&(s.nrDev=!0))}catch(c){}s.nrDev&&i.on("internal-error",function(t){r(t.st...

```

</details>

**Comment by okiehsch** [comment]
This is weird, do you get Instart Logic in the source of `metacritic.com`?
![image](https://user-images.githubusercontent.com/16838044/64338060-03366b00-cfe1-11e9-91e7-0bbfe42d6d3b.png)

**Comment by llacb47** [comment]
I think it might depend on user agent if they serve the IL script.

**Comment by gwarser** [comment]
![image](https://user-images.githubusercontent.com/886325/64347510-5c5bca00-cff4-11e9-9ff4-9c841991a91e.png)

**Comment by rpodric** [comment]
@okiehsch 

I'll try to catch up:

-I don't know if the redirects were happening last night, but this morning they are.
-This morning, your "//Copyright Instart Fri May 24 2019 - All rights reserved - version: 10.6.32" string appears on tvguide.com, as if by magic, when it didn't through multiple reloads last night. Both times, however, last night and now, the site looks and feels as if ad blocking is completely disabled on it. I know it isn't, by looking at the uBO panel, but if I couldn't look at that I would have guessed that it was disabled.
-Yes, also seeing the very same Instart string on Metacritic.

**Comment by gorhill** [comment]
`//Copyright Instart Fri May 24 2019` does not tell which version of IL crap is embedded. There is a new version that broke sites which were listed in uBO-Extra, so they had to be removed.

What is needed is dev work to be undertaken to understand how the new IL crap work, and if it can be defused. At some point, that developer will have to be someone else than me, I do not have infinite time and interest to spend on this. I do not visit any IL-backed site personally, so the motivation to work on this beside everything else I already have to maintain because I would be affected myself is not that great.

**Comment by okiehsch** [comment]
@rpodric 
You asked after the removal of sites like `tvguide.com` what should be the course of action?

On your end you can

- Use Firefox for sucgg sites
- Disable javascript, This will break some stuff but should be good enough if you only want to read articles.
-  Sideload an old version of `uBO-extra`. I checked `tvguide.com` with `uBO-Extra-2.83` and it works fine without any redirection or site breakage on my end.

**Comment by llacb47** [comment]
https://github.com/uBlockOrigin/uAssets/issues/6260#issuecomment-529144977

**Comment by llacb47** [comment]
Can these filters be added for IGN? @okiehsch 
```
||static.doubleclick.net/instream/ad_status.js$script,domain=ign.com,redirect=noop.js
||imasdk.googleapis.com/js/sdkloader/ima3.js$script,domain=ign.com,redirect=noop.js
||g01.ign.com^$script,redirect=noop.js
||g02.ign.com^$script,redirect=noop.js
```

**Comment by okiehsch** [comment]
For `ign.com` see https://github.com/uBlockOrigin/uAssets/commit/33a403e98aa4f343c012936d2bdda0cca8f76319#commitcomment-34805881

**Comment by llacb47** [comment]
Please test/add the following for `ratemyprofessors.com`:

```
! same rule as metacritic, could be made generic?
||ratemyprofessors.com^*/ad?i10c.getabdurl=*&ad_channel=$media,1p,redirect=noop.txt
||ratemyprofessors.com/g00/*$ping
```

Also for metacritic, the rule `||metacritic.com/g00/*$ping` can be added as well so that no g00 content is loaded at all

**Comment by okiehsch** [comment]
I will add your filters to `uBO-filters`-
I have re-added some sites to `uBO-extra` https://github.com/gorhill/uBO-Extra/commit/448a3ab5133b1abcc852df91660ad2a5563d55c1
So those sites should work fine  as soon as the next `uBO-extra` version is released.

**Comment by llacb47** [comment]
https://github.com/uBlockOrigin/uAssets/commit/1f28b7fc1fd8df07b77c381311b9936b7d605d23

**Comment by llacb47** [comment]
@okiehsch please add `@@||tvguide.com^$generichide` for `tvguide.com`

**Comment by llacb47** [comment]
And `@@||maxpreps.com^$generichide`

**Comment by llacb47** [comment]
Another one: `@@||metrolyrics.com^$generichide` for `metrolyrics.com`

**Comment by llacb47** [comment]
Oh wait, some of these sites are being added back to uBo extra. https://github.com/gorhill/uBO-Extra/commit/448a3ab5133b1abcc852df91660ad2a5563d55c1

**Comment by mapx-** [comment]
It does not matter, still useful for those not using uBo extra

**Comment by llacb47** [comment]
@okiehsch https://github.com/NanoMeow/QuickReports/issues/1923#issuecomment-535641806

**Comment by llacb47** [comment]
Instart is back on `msn.com`.

Fix:

```
*$media,redirect=noop.txt,domain=msn.com,3p
@@||eus-streaming-video-msn-com.akamaized.net^$media,domain=msn.com
```

**Comment by okiehsch** [comment]
How about
````
baggersmag.com,cbssports.com,goal.com,msn.com,thoughtcatalog.com##+js(aopw, Object.getOwnPropertyDescriptor)
````

**Comment by gorhill** [comment]
That scriptlet can't accomplish anything -- by the time you find a `script` tag, its javascript content has already been executed, there is no undoing this by removing the tag.

**Comment by uBlock-user** [comment]
> That scriptlet can't accomplish anything -- by the time you find a script tag, its javascript content has already been executed, there is no undoing this by removing the tag.

On the contrary, it works as expected on `sportingnews.com`

**Comment by gorhill** [comment]
It does work, not sure why -- my understanding is that JS is executed as soon as the browser add the`script` node to the DOM, which is the case here since you could find it using `querySelectorAll`. I wish a browser dev familiar with HTML parsing would shed light on this.

**Comment by uBlock-user** [comment]
Not just this one, element remover scriptlet that I use also has the same base and that works perfectly as expected and elements that I remove from DOM don't flash before getting removed.

**Comment by gorhill** [comment]
> don't flash before getting removed

Not unexpected since you are removing them directly from the mutation observer. uBO defers applying `display: none;` outside the mutation observer because this could easily become a performance issue given how some pages continuously modify the DOM.

But I would be open to have a generic scriptlet to remove elements as you do in your scriptlet based on both a selector and a text needle.

**Comment by uBlock-user** [comment]
> because this could easily become a performance issue given how some pages continuously modify the DOM.

Do you see any performance issue in my scriptlet here as I'm not disconnecting the MO ?

> But I would be open to have a generic scriptlet to remove elements as you do in your scriptlet based on both a selector and a text needle.

No idea what would I look for as a text needle, if you know, you should implement it already.

**Comment by gorhill** [comment]
A bit more thoughts... Maybe I could just implement a poor's man HTML filtering for Chromium-based browser (not guaranteed to work though), which would essentially do what you do in your scriptlet except it would uninstall itself once the DOM content has been loaded. This means maybe the following filter would work without needing any new scriptlet:

    sportingnews.com##^script:has-text(/INSTART/)

**Comment by jspenguin2017** [comment]
`MutationObserver` can cancel and alter inline scripts on Chromium and legacy Edge. Possibly other browsers too, but I have yet to test that. 

For Firefox, removing inline script tags does not cancel their execution, but overwriting `textContent` will change the code that would execute. So changing `textContent` to an empty script would effectively cancel the execution. Note that Firefox has native `onbeforescriptexecute` which can be used to cancel inline scripts. 

I am not too sure why it behaves like this, I highly doubt it is intentional, but it looks like canceling inline scripts execution have been fairly stable over the years. However, `textContent` attribute does not always work inside the mutation event handler, so altering inline scripts can sometimes be challenging. 

See example implementation: 
https://github.com/jspenguin2017/Snippets/blob/master/onbeforescriptexecute.html

**Comment by llacb47** [comment]
@okiehsch please add
```
||www.googletagservices.com/activeview/js/current/osd.js$script,redirect=noop.txt,domain=webmd.com
||www.googletagservices.com/tag/js/gpt.js$script,redirect=noop.txt,domain=webmd.com
||z.moatads.com/nativo483ETsY34/moatad.js$script,redirect=noop.txt,domain=webmd.com
||c.amazon-adsystem.com/aax2/apstag.js$script,redirect=noop.txt,domain=webmd.com
||securepubads.g.doubleclick.net/gpt/pubads_impl_2019032901.js$script,redirect=noop.txt,domain=webmd.com
```

for WebMD Instart: `https://www.webmd.com/heart-disease/qa/how-does-angiography-help-in-diagnosing-atherosclerosis`

**Comment by okiehsch** [comment]
Thanks, adding
`*$script,domain=webmd.com,3p,redirect-rule=noopjs`
is "easier". 😉 
https://github.com/gorhill/uBlock/wiki/Static-filter-syntax#redirect-rule

**Comment by llacb47** [comment]
Well, duh, I know that, but that breaks videos and probably many other things.

**Comment by okiehsch** [comment]
How? That filter does not block anything it only redirects all scripts that are **already** blocked by existing filters.
Read the wiki link I posted.

**Comment by llacb47** [comment]
Yeah, I read it too fast

**Comment by llacb47** [comment]
Another issue... `https://www.edmunds.com` is broken with uBO Extra enabled. To use the site, uBO Extra must be disabled and the rule `||cdn.speedcurve.com/js/lux.js$script,domain=edmunds.com,redirect=noop.js` must also be added.

**Comment by okiehsch** [comment]
![image](https://user-images.githubusercontent.com/16838044/67062134-c83c6100-f162-11e9-8adc-e73522538e18.png)
Does not work on my end, I still get redirected.

**Comment by okiehsch** [comment]
The Instart Logic script at that site is slightly different than the other ones.
````
edmunds.com##+js(aopw, ed)
edmunds.com##+js(aopw, i10cdone)
````
works for me. If it does not work for you I can add your redirection as well, can't hurt.

**Comment by okiehsch** [comment]
The videos at `webmd.com` are indeed all broken by existing filters.

**Comment by mapx-** [comment]
It's uBo-extra breaking the video

**Comment by okiehsch** [comment]
I get the endless "Video Loading" message with `uBO-extra` disabled.

**Comment by mapx-** [comment]
try (uBo extra disabled)
```
@@||securepubads.g.doubleclick.net/$script,domain=webmd.com
*/videoplayback?expire$media,redirect=noopmp3-0.1s,domain=webmd.com
```

**Comment by okiehsch** [comment]
Does not work for me, I mentioned that 

> I can't get them to play even if I allow all necessary requests

````
@@||doubleclick.net/$script,domain=webmd.com
@@||imasdk.googleapis.com/js/sdkloader/ima3.js$script,domain=webmd.com
@@||doubleclick.net/gampad/ads*=www.webmd.com$xhr,domain=imasdk.googleapis.com
````
should allow the player to work, atleast the video ad should show but I am still stuck with an endless loading message.
![image](https://user-images.githubusercontent.com/16838044/67075146-e0c07180-f18a-11e9-9593-f223765cdf29.png)

**Comment by mapx-** [comment]
working fine for me, in chrome, european ip, uBo extra disabled only th filters above

**Comment by okiehsch** [comment]
Well, I can remove `webmd` from `uBO-extra`, the `uAssets` filter works fine on my end.
The video player is broken with a US or EU IP on my end.

**Comment by mapx-** [comment]
I'll add your filters + my media one

**Comment by okiehsch** [comment]
I think the reason is the `redirect-rule` filter does it work on your end if you add
````
||googletagservices.com/activeview/js/current/osd.js$script,redirect=noopjs,domain=webmd.com
||googletagservices.com/tag/js/gpt.js$script,redirect=noopjs,domain=webmd.com
||moatads.com/*/moatad.js$script,redirect=noopjs,domain=webmd.com
||amazon-adsystem.com/aax2/apstag.js$script,redirect=noopjs,domain=webmd.com
||securepubads.g.doubleclick.net/gpt/pubads_impl_$script,redirect=noopjs,domain=webmd.com
*$script,domain=webmd.com,3p,redirect-rule=noopjs,badfilter
````

**Comment by mapx-** [comment]
yes, it works. The second filter is not necessary but we still need (already in unbreak list)
`@@||doubleclick.net/$script,domain=webmd.com`

**Comment by okiehsch** [comment]
No, the redirections all occur, you can see it in the browser console and also if they would not happen you would get redirected by the Instart Logic script.
Redirecting all blocked scripts somehow causes the video player to get stuck.

**Comment by mapx-** [comment]
So, it means redirecting a certain script breaks the video .. but the same script completely blocked does not break the video, really weird

**Comment by mapx-** [comment]
Maybe in case the script is blocked they are testing it and take the right way. Instead - in case the script exists (even redirected) they are taking another way expecting some results => breaking the video

**Comment by llacb47** [comment]
I don't think the filters
```
@@||doubleclick.net/$script,domain=webmd.com
@@||imasdk.googleapis.com/js/sdkloader/ima3.js$script,domain=webmd.com
@@||doubleclick.net/gampad/ads*www.webmd.com$xhr,domain=imasdk.googleapis.com
```

are needed anymore (The first one is fixed in easylist).
Test here `https://www.webmd.com/diet/video/truth-about-drinking-water`

**Comment by llacb47** [comment]
https://github.com/uBlockOrigin/uAssets/issues/6568#issuecomment-569181914

**Comment by okiehsch** [comment]
`msn.com` works fine with `uBO-extra` on my end.

**Comment by morgan357** [comment]
ENGAÑA AL DETECTOR DE BLOQUEADORES, PUEDO VER LA PÁGINA, PERO NO PUEDE ACCEDER A LOS RECURSOS DE MULTIMEDIA.  https://esradio.libertaddigital.com/?utm_source=&utm_medium=RADIO&utm_campaign=menu

**Comment by mapx-** [comment]
add
`@@||cdn.adswizz.com/adswizz/js/SynchroClient2.js$script,domain=esradio.libertaddigital.com`

uBO filter issues to be reported here:
https://github.com/uBlockOrigin/uAssets/issues

**Comment by Neurotoxin001** [comment]
Is this extension still relevant? Been using it for wowhead.com but now I can't see this flood with ad network requests even with uBO-Extra disabled.

**Comment by uBlock-user** [comment]
For Instart Logic ? No. The company got sold out to Akamai and they ceased their operations since then. Their homepage is also gone.

**Comment by llacb47** [comment]
> The company got sold out to Akamai

`https://www.akamai.com/us/en/about/news/press/2020-press/akamai-technologies-acquires-customers-and-select-intellectual-property-from-instart.jsp`
